calyx-opt/src/passes/static_promotion.rs

```diff
@@ -733,14 +983,46 @@ impl Visitor for StaticPromotion {
     fn finish_if(
         &mut self,
         s: &mut ir::If,
-        _comp: &mut ir::Component,
-        _sigs: &LibrarySignatures,
+        comp: &mut ir::Component,
+        sigs: &LibrarySignatures,
         _comps: &[ir::Component],
     ) -> VisResult {
-        if let Some(sif) = s.make_static() {
-            return Ok(Action::change(ir::Control::Static(
-                ir::StaticControl::If(sif),
-            )));
+        let mut builder = ir::Builder::new(comp, sigs);
+        if Self::can_be_promoted(&s.tbranch)
+            && Self::can_be_promoted(&s.fbranch)
+        {
+            // Both branches can be promoted
+            let approx_size_if = Self::approx_size(&s.tbranch)
+                + Self::approx_size(&s.fbranch)
+                + APPROX_IF_SIZE;
```

Comment:
Can we encapsulate this computation in a function that takes `&ir::Control` and returns a `u64` or something? Would make the logic centralized and easier to understand.

Reply:
Yeah I have a function that does this (i.e., takes `&ir::Control` and returns a `u64`). The only thing is that this is in `visit_if` with takes in a `& mut ir::StaticIf`. I definitely think it would be cleaner if we could always use just one function for everything, though...

Reply:
Ah, that's annoying (and I guess something that often happens in the IR)